From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tristan Dunn <hello@tristandunn.com>
Date: Thu, 27 Aug 2026 19:32:10 -0400
Subject: Lower the maximum scroll acceleration.

Sequential scroll motions accelerate down to a divisor of 1, which is far
faster than macOS and overshoots easily. Clamp the divisor to
SCROLL_ACCEL_MIN instead.
---
 hid-magicmouse.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hid-magicmouse.c b/hid-magicmouse.c
index 60ff889..70734f5 100644
--- a/hid-magicmouse.c
+++ b/hid-magicmouse.c
@@ -86,6 +86,7 @@ MODULE_PARM_DESC(scroll_while_clicking, "Enable scrolling when a button is held"
 #define SCROLL_HR_MULT (120 / SCROLL_HR_STEPS)
 #define SCROLL_HR_THRESHOLD 90 /* units */
 #define SCROLL_ACCEL_DEFAULT 7
+#define SCROLL_ACCEL_MIN 3
 
 /* Touch surface information. Dimension is in hundredths of a mm, min and max
  * are in units. */
@@ -305,7 +306,8 @@ static void magicmouse_emit_touch(struct magicmouse_sc *msc, int raw_id, u8 *tda
 			if (scroll_acceleration && time_before(now,
 						msc->scroll_jiffies + HZ / 2))
 				msc->scroll_accel = max_t(int,
-						msc->scroll_accel - 1, 1);
+						msc->scroll_accel - 1,
+						SCROLL_ACCEL_MIN);
 			else
 				msc->scroll_accel = SCROLL_ACCEL_DEFAULT;
 
